IPython/core/debugger.py

```diff
@@ -621,3 +621,14 @@ def do_where(self, arg):
                 self.print_stack_trace()
 
         do_w = do_where
+
+
+def set_trace(frame=None, condition=True):
```

Comment:
Let's stick to the same API as `pdb` and `IPython.terminal.debugger`, with `set_trace()` taking no parameters. We can discuss adding parameters as a separate step, and if we do we'll want to keep `core` and `terminal` in sync.